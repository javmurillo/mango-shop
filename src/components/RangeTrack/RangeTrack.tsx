import React, { Component } from 'react';
import styled from 'styled-components';
import { RangeTrackProps } from './range-track-props.interface';

const StyleTrack = styled.div`
  position: absolute;
  width: 100%;
  margin-top: 11px;
  border-radius: 10%;
  background-color: white;
  border: 3px solid black;
`;

export default class RangeTrack extends Component<RangeTrackProps> {
  render(): JSX.Element {
    const { trackRef } = this.props;
    return <StyleTrack ref={trackRef} />;
  }
}
