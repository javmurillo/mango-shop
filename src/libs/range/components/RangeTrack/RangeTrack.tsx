import React from 'react';
import styled from 'styled-components';
import { RangeTrackProps } from './range-track-props.interface';

const StyleTrack = styled.div`
  position: absolute;
  width: 100%;
  margin-top: 11px;
  border-radius: 10%;
  background-color: white;
  border: 3px solid black;
`;

/**
 * RangeTrack component
 * @param props RangeTrackProps
 */
export const RangeTrack = (props: RangeTrackProps) => {
  return <StyleTrack ref={props.trackRef} aria-label={props.ariaLabel} />;
};
