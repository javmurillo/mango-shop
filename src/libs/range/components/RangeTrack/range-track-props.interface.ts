export interface RangeTrackProps {
  trackRef?: (track: HTMLDivElement) => void;
  ariaLabel?: string;
}
