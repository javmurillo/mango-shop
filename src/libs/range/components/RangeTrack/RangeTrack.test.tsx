import { cleanup, render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { RangeTrack } from './RangeTrack';

afterEach(cleanup);

const setup = () => {
  const utils = render(<RangeTrack ariaLabel="track" />);
  const rangeTrack = utils.getByLabelText('track');
  return {
    rangeTrack,
    ...utils,
  };
};

describe('<RangeTrack/> tests', () => {
  test('renders correctly', () => {
    const { rangeTrack } = setup();
    expect(rangeTrack).toBeInTheDocument();
  });

  test('shoud match snapshot', () => {
    const testRendererJson = TestRenderer.create(<RangeTrack />).toJSON();
    expect(testRendererJson).toMatchSnapshot();
  });
});
