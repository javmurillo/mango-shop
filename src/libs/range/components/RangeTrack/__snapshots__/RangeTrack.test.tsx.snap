// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeTrack/> tests shoud match snapshot 1`] = `
<div
  className="sc-bdfBwQ cIBWvr"
/>
`;
